run/test/specs/disappearing_message_checks_android.spec.ts

```diff
@@ -0,0 +1,429 @@
+      strategy: "accessibility id",
+      selector: "Media message",
+    }),
+  ]);
+  await closeApp(device1, device2);
+}
+
+async function disappearingVoiceMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  await newContact(platform, device1, userA, device2, userB);
+  await setDisappearingMessage(device1, [
+    "1o1",
+    "Disappear after send option",
+    "10 seconds",
+  ]);
+  await device1.longPress("New voice message");
+  await device1.clickOnElement("Continue");
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.ScrollView/android.widget.LinearLayout/android.widget.LinearLayout/android.widget.LinearLayout[2]/android.widget.Button[1]`
+  );
+  await device1.pressAndHold("New voice message");
+  // await device1.clickOnElement("OK");
+  // await device1.pressAndHold("New voice message");
+  await device1.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await device2.clickOnElement("Untrusted attachment message");
+  await sleepFor(200);
+  await device2.clickOnElement("Download");
+  await sleepFor(10000);
+  await device1.hasElementBeenDeletedNew({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await device2.hasElementBeenDeletedNew({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await closeApp(device1, device2);
+}
+
+async function disappearingGifMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const testMessage = "Testing disappearing messages for GIF's";
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  await newContact(platform, device1, userA, device2, userB);
+  await setDisappearingMessage(device1, [
+    "1o1",
+    "Disappear after send option",
+    "10 seconds",
+  ]);
+  await device1.navigateBack(platform);
+  // Click on attachments button
+  await device1.clickOnElement("Attachments button");
+  // Select GIF tab
+
+  await device1.clickOnElement("GIF button");
+  await device1.clickOnElementAll({
+    strategy: "accessibility id",
+    selector: "Continue",
+  });
+  // Select gif
+  await sleepFor(3000);
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.RelativeLayout/android.widget.ScrollView/androidx.viewpager.widget.ViewPager/android.widget.FrameLayout/androidx.recyclerview.widget.RecyclerView/android.widget.FrameLayout[1]`
+  );
```

Comment:
those xpath in the code are not pretty.
we could make the XPath type be an enum instead, so when calling clickOnEleemntXpath you could do

`clickOnElementXpath(XPath.anExplicitName)`
what do you think?